### ISSUE TITLE
Rancher shell can't start with broken policy settings

### ISSUE THREAD

**Issue by kravciak** [kubewarden/namespace-label-propagator-policy#8]
### To reproduce
Create new policy (context-aware settings are broken in current UI, use them to break policy)
-> new ClusterAdmissionPolicy
-> Namespace label propagator (Protect mode)
-> fill name, add some label in settings

Wait until policy is active, then start kubectl shell. My UI generated this YAML:

<details><summary>Generated YAML</summary>

```yaml
apiVersion: policies.kubewarden.io.v1
kind: ClusterAdmissionPolicy
metadata:
  name: nlp
  namespace: ''
  annotations:
    artifacthub/pkg: namespace-label-propagator-policy/namespace-label-propagator-policy/0.1.2
spec:
  policyServer: default
  module: ghcr.io/kubewarden/policies/namespace-label-propagator:v0.1.2
  rules:
    - apiGroups:
        - ''
      apiVersions:
        - v1
      resources:
        - replicationcontrollers
        - pods
      operations:
        - CREATE
        - UPDATE
    - apiGroups:
        - apps
      apiVersions:
        - v1
      resources:
        - deployments
        - replicasets
        - statefulsets
        - daemonsets
      operations:
        - CREATE
        - UPDATE
    - apiGroups:
        - batch
      apiVersions:
        - v1
      resources:
        - jobs
        - cronjobs
      operations:
        - CREATE
        - UPDATE
  contextAware: false
  mutating: true
  settings:
    description: null
    propagatedLabels:
      - some-label
  mode: protect
ignoreRancherNamespaces: true
```
</details>

### Current result
kubectl shell does not start, required pod in cattle-system namespace is not created. Policyserver logs this message when I start kubectl shell:

```
2023-06-02T14:41:02.754557Z ERROR validation{host="policy-server-default-5895c59fc-dqqvm" policy_id="clusterwide-nlp" kind="Pod" kind_group="" kind_version="v1" name="" namespace="cattle-system" operation="CREATE" request_uid="7049580d-1247-4454-8f32-9e5d138c6ee1" resource="pods" resource_group="" resource_version="v1" subresource=""}:policy_eval:validate{self=PolicyEvaluator { id: "clusterwide-nlp", settings: {"propagatedLabels": Array [String("some-label")]} }}: policy_evaluator::runtimes::wapc: Policy tried to access a Kubernetes resource it doesn't have access to policy="clusterwide-nlp" resource_requested="v1/Namespace" resources_allowed={}
```

![Screenshot from 2023-06-02 16-37-05](https://github.com/kubewarden/namespace-label-propagator-policy/assets/538604/0faae39d-a3c2-4916-9312-5b87c07a9790)

### Expected result

Policy should ignore rancher resources even when it fails to start.
Rancher should be allowed to start a pod that is then connected to the UI.

```bash
~ k get pods -n cattle-system -o name
pod/dashboard-shell-69j7z
```

**Comment by jvanz** [comment]
As far I can understand, the yaml you have shared is the yaml from Rancher UI before apply the policy in the cluster. Am I right? 

Can you get the yaml definition of the policy after it is running in the cluster? 

```console
kubectl get clusteradmissionpolicies  nlp -o yaml
```


The field `ignoreRancherNamespaces` should be translated into  `namespaceSelector` field. This is an example of my local test cluster:

```yaml
# output of the k8s get clusteradmissionpolicies  nlp -o yaml 
apiVersion: policies.kubewarden.io/v1
kind: ClusterAdmissionPolicy
metadata:
  annotations:
    artifacthub/pkg: namespace-label-propagator-policy/namespace-label-propagator-policy/0.1.2
  creationTimestamp: '2023-06-02T19:44:24Z'
  finalizers:
    - kubewarden
  generation: 2
  managedFields:
    - apiVersion: policies.kubewarden.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:artifacthub/pkg: {}
        f:spec:
          .: {}
          f:mode: {}
          f:module: {}
          f:mutating: {}
          f:namespaceSelector:
            .: {}
            f:matchExpressions: {}
          f:policyServer: {}
          f:rules: {}
          f:settings:
            .: {}
            f:description: {}
            f:propagatedLabels: {}
          f:timeoutSeconds: {}
      manager: rancher
      operation: Update
      time: '2023-06-02T19:50:36Z'
    - apiVersion: policies.kubewarden.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:conditions:
            .: {}
            k:{"type":"PolicyActive"}:
              .: {}
              f:lastTransitionTime: {}
              f:message: {}
              f:reason: {}
              f:status: {}
              f:type: {}
            k:{"type":"PolicyServerConfigurationUpToDate"}:
              .: {}
              f:lastTransitionTime: {}
              f:message: {}
              f:reason: {}
              f:status: {}
              f:type: {}
            k:{"type":"PolicyUniquelyReachable"}:
              .: {}
              f:lastTransitionTime: {}
              f:message: {}
              f:reason: {}
              f:status: {}
              f:type: {}
          f:mode: {}
          f:policyStatus: {}
      manager: manager
      operation: Update
      subresource: status
      time: '2023-06-02T19:51:29Z'
  name: nlp
  resourceVersion: '15247'
  uid: a181381c-1ced-47dd-bb37-cd03861da995
spec:
  mode: protect
  module: ghcr.io/kubewarden/policies/namespace-label-propagator:v0.1.2
  mutating: true
  namespaceSelector:
    matchExpressions:
      - key: kubernetes.io/metadata.name
        operator: NotIn
        values:
          - calico-system
          - cattle-alerting
          - cattle-fleet-local-system
          - cattle-fleet-system
          - cattle-global-data
          - cattle-global-nt
          - cattle-impersonation-system
          - cattle-istio
          - cattle-logging
          - cattle-pipeline
          - cattle-prometheus
          - cattle-system
          - cert-manager
          - ingress-nginx
          - kube-node-lease
          - kube-public
          - kube-system
          - rancher-operator-system
          - security-scan
          - tigera-operator
  policyServer: default
  rules:
    - apiGroups:
        - ''
      apiVersions:
        - v1
      operations:
        - CREATE
        - UPDATE
      resources:
        - replicationcontrollers
        - pods
    - apiGroups:
        - apps
      apiVersions:
        - v1
      operations:
        - CREATE
        - UPDATE
      resources:
        - deployments
        - replicasets
        - statefulsets
        - daemonsets
    - apiGroups:
        - batch
      apiVersions:
        - v1
      operations:
        - CREATE
        - UPDATE
      resources:
        - jobs
        - cronjobs
  settings:
    description: null
    propagatedLabels:
      - some-label
  timeoutSeconds: 10
status:
  conditions:
    - lastTransitionTime: '2023-06-02T19:51:29Z'
      message: The policy webhook has been created
      reason: PolicyActive
      status: 'True'
      type: PolicyActive
    - lastTransitionTime: '2023-06-02T19:51:28Z'
      message: The latest replica set is uniquely reachable
      reason: LatestReplicaSetIsUniquelyReachable
      status: 'True'
      type: PolicyUniquelyReachable
    - lastTransitionTime: '2023-06-02T19:44:24Z'
      message: Configuration for this policy is up to date
      reason: ConfigurationVersionMatch
      status: 'True'
      type: PolicyServerConfigurationUpToDate
  mode: protect
  policyStatus: active
```

I would like to check the final yaml applied into the cluster. Because in my local cluster I could not simulate this issue yet.

**Comment by kravciak** [comment]
I tried to reproduce and problem is still there:
- deploy v2.7.5-rc5 rancher, 1.0.5 kubewarden UI, k3d -> k3s v1.25.8 +k3s1
- enable default policy server + recommended policies in monitor mode
- add namespace label propagator policy
  - ignore rancher namespaces: true
  - mode: protect
  - settings -> add label -> lbl
  - go to `Edit YAML` tab and remove `contextAwareResources` definition

After policy is active I can't start kubectl shell, namespaceSelector is not present in `kubectl get clusteradmissionpolicies  nlp -o yaml`

**Comment by jvanz** [comment]
Thinking more about this... I think this is a UI issue. The Helm chart and the controller do not translate the "ignore rancher namespaces" into namespace selector configuration. This is done by the UI. Let try to find out where this is done. 

FYI: @jordojordo

**Comment by jvanz** [comment]
> I guess these come from the UI. Am I right?

Yes. AFAICS, this is a checkbox from the UI which should be replaced by the namespace selector configuration where all the Rancher namespaces will be ignored by the policy. But for some reason, this is not done.

**Comment by jordojordo** [comment]
@jvanz You are correct with [this location](https://github.com/kubewarden/ui/blob/572f2efb15c507f3d3266722c5ccbe895b240c00/pkg/kubewarden/components/Policies/Create.vue#L247-L250) that should be adding the namespace selector. Seems a simple fix move that statement below the next condition to apply the property after the Yaml has been parsed. I've created an issue on the UI: https://github.com/kubewarden/ui/issues/399

**Comment by jvanz** [comment]
This is fixed in the latest UI `v1.0.6`. I can start the rancher shell and execute `kubectl` commands. Closing issue.

**Comment by kravciak** [comment]
Confirmed, works as expected.
